cpp/oneapi/dal/test/engine/dataframe_common.cpp

```diff
@@ -344,25 +381,75 @@ static homogen_table build_homogen_table(Policy& policy,
 }
 
 template <typename Policy>
-static table build_table(Policy& policy, const dataframe& df, const table_id& id) {
+static homogen_table build_homogen_table(Policy& policy,
+                                         const array<float>& data,
+                                         const table_id& id,
+                                         std::int64_t row_count,
+                                         std::int64_t first_column,
+                                         std::int64_t last_column) {
+    const std::int64_t data_column_count = data.get_count() / row_count;
+    const std::int64_t column_count = last_column - first_column;
+
+    if (first_column == 0 && data_column_count == column_count) {
+        return wrap_to_homogen_table(policy, data, id, row_count, column_count);
+    }
+    else {
+        auto arr = array<float>::empty(row_count * column_count);
+        float* arr_ptr = arr.get_mutable_data();
+        const auto data_ptr = data.get_data();
```

Comment:
```suggestion
        float* arr_ptr = arr.get_mutable_data();
        const float* data_ptr = data.get_data();
```

Sometimes it's hard to understand `auto` type from the context, it's helpful for reader to understand if the subsequent loop converts types.